ion-schema/src/main/kotlin/com/amazon/ionschema/internal/ConstraintFactory.kt

```diff
@@ -25,7 +25,7 @@ internal interface ConstraintFactory {
     /**
      * If [name] is a recognized constraint name, returns `true`, otherwise `false`.
      */
-    fun isConstraint(name: String): Boolean
+    fun isConstraint(name: String, schema: Schema): Boolean
```

Comment:
Can we just pass the Isl version here instead of passing the entire schema?

Reply:
We could. I made an arbitrary decision for `isConstraint` and `constraintFor` to have the same parameters, but I can change it if you think it's worth doing.

Reply:
I see. I don't think its a hard requirement to make this change, it could work either way. But I just wanted to make it so that it only requires the value that is needed for `isConstraint` implementation rather than entire `schema`.